akka-distributed-data/src/main/scala/akka/cluster/ddata/GSet.scala

```diff
@@ -53,15 +55,39 @@ final case class GSet[A](elements: Set[A]) extends ReplicatedData with Replicate
   /**
    * Adds an element to the set
    */
-  def add(element: A): GSet[A] = assignAncestor(copy(elements + element))
+  def add(element: A): GSet[A] = {
+    val newDelta = _delta match {
+      case Some(e) ⇒ Some(new GSet(e.elements + element))
+      case None    ⇒ Some(new GSet[A](Set.apply[A](element)))
```

Comment:
Yes, everywhere _outside_ of `GSet.scala` as here the curried constructor is still visible and would clash with the single argument `apply` from Companion Object.